src/Compilers/CSharp/Portable/Symbols/AsyncKind.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal enum AsyncKind : byte
+    {
+        NotAsync = 0,
+        Invalid,
+        Void,
+        Task,
```

Comment:
I actually liked the short form `Task` to mean `Task` or `Task`-like (similarly for `Task_T`), particularly since the consumers of this enum are typically not interested in the distinction. Any suggestions on names? `TaskOrTasklike` seems clumsy.